### ISSUE TITLE
Determining pose of glass objects

### ISSUE THREAD

**Issue by dj4park** [NVlabs/Deep_Object_Pose#65]
Has there been any experiment done with using DOPE to determine the pose of a glass bottle? If not, would there be any reason as to why it would not work?

Currently I am experimenting with a Soju bottle. I have trained the network using 6000 basic NDDS images for 60 epochs using a batch size of 4 and a learning rate of 0.001 but I am having difficulties in detection. 

Could the problem be due to the hyperparameters, image quantity/quality? Or could it be due to the texture of the glass bottle?

**Comment by TontonTremblay** [comment]
Do you have symmetries on you object, could you show me some training examples and testing examples? I never tried transparent objects with DOPE.

**Comment by dj4park** [comment]
Following a similar strategy to @Abdul-Mukit and his [video](https://www.youtube.com/watch?v=g1adPMSmrXY), I created 6000 simple images using NDDS and trained the model for 60 epochs on a NVIDIA Quadro P3200 Mobile setting the learning rate to 0.001 and the batch size to 4.

Here is a photo of the object that I created using Blender.
![image](https://user-images.githubusercontent.com/25046990/63099415-0fc82600-bf43-11e9-9088-0e8bda79a130.png)

And here are some training examples that I made with NDDS.
![000010 depth](https://user-images.githubusercontent.com/25046990/63099538-443be200-bf43-11e9-8574-76f7f3551876.png)
![000010 is](https://user-images.githubusercontent.com/25046990/63099539-443be200-bf43-11e9-9260-677260145258.png)
![000010](https://user-images.githubusercontent.com/25046990/63099540-44d47880-bf43-11e9-8bac-7b46dc0c7379.png)
![000010 cs](https://user-images.githubusercontent.com/25046990/63099541-44d47880-bf43-11e9-9e9a-1c3aec39d8ac.png)

And here is the actual object itself with which I am testing the model.
![soju1](https://user-images.githubusercontent.com/25046990/63100706-a269c480-bf45-11e9-988a-2bce28991028.JPG)
![soju2](https://user-images.githubusercontent.com/25046990/63100708-a4338800-bf45-11e9-8ede-e038efa82577.JPG)

**Comment by Abdul-Mukit** [comment]
The contributors/authors will give the best suggestions, but as I am working on things similar to this I would like to share some opinion. Your 3D model doesn't look as transparent as the original bottle, the labels don't match either. The smoothness of the surface is off I think. I think you should edit the material, and increase metalic property and decrease roughness. You should also work on making the model more transparent. The cap color doesn't match either. The label on the neck of the bottle is missing in your model. The cap doesn't have the same material as the glass and the mesh of the cap doesn't look like the original cap. For the labels of the bottle, you can scan the label image and use it as a texture for the 3D model. You may use u-v mapping to do that - not sure if that will work but I would have tried that. Also, try randomizing the background with images. In my small experience, I have felt background randomization with real images tends to prevent overfitting better compared to the small number of abstract background generated by NDDS.

**Comment by TontonTremblay** [comment]
I second!

**Comment by aditya2592** [comment]
Hi @dj4park. Were you able to successfully train for your custom object?

**Comment by dj4park** [comment]
I am randomizing the background with the default images provided with NDDS.

**Comment by sanjaysswami** [comment]
> Following a similar strategy to @Abdul-Mukit and his [video](https://www.youtube.com/watch?v=g1adPMSmrXY), I created 6000 simple images using NDDS and trained the model for 60 epochs on a NVIDIA Quadro P3200 Mobile setting the learning rate to 0.001 and the batch size to 4.
> 
> Here is a photo of the object that I created using Blender.
> ![image](https://user-images.githubusercontent.com/25046990/63099415-0fc82600-bf43-11e9-9088-0e8bda79a130.png)
> 
> And here are some training examples that I made with NDDS.
> ![000010 depth](https://user-images.githubusercontent.com/25046990/63099538-443be200-bf43-11e9-8574-76f7f3551876.png)
> ![000010 is](https://user-images.githubusercontent.com/25046990/63099539-443be200-bf43-11e9-9260-677260145258.png)
> ![000010](https://user-images.githubusercontent.com/25046990/63099540-44d47880-bf43-11e9-8bac-7b46dc0c7379.png)
> ![000010 cs](https://user-images.githubusercontent.com/25046990/63099541-44d47880-bf43-11e9-9e9a-1c3aec39d8ac.png)
> 
> And here is the actual object itself with which I am testing the model.
> ![soju1](https://user-images.githubusercontent.com/25046990/63100706-a269c480-bf45-11e9-988a-2bce28991028.JPG)
> ![soju2](https://user-images.githubusercontent.com/25046990/63100708-a4338800-bf45-11e9-8ede-e038efa82577.JPG)

How did you insert that exact texture (green color and written things etc) in blender? Rather can you please explain me how did you create this 3d model? I a also trying to generate the dataset for my own object but I don't know how to add textures. If you can help me with this issue it would be really greate help to me.
